project/CRUD/roles.php

```diff
@@ -1,3 +1,173 @@
-<main>
-    <p>This is Roles</p>
-</main>
\ No newline at end of file
+<?php
+
+// initializing variables
+$id = $role_name = "";
+
+// ADD
+if (isset($_POST['add_role'])) {
+
+    // receive all input values from the form
+    $role_name = mysqli_real_escape_string($conn, $_POST['role_name']);
+
+    // form validation: ensure that the form is correctly filled ...
+    // by adding (array_push()) corresponding error unto $errors array
+    if (empty($role_name)) {
+        array_push($errors, "Role Name is required");
+    }
+
+    if (count($errors) == 0) {
+        $add = "INSERT INTO roles (role_name) VALUES('$role_name');";
+
+        if (mysqli_query($conn, $add)) {
+            array_push($success, "role added Successful");
+            // clear variables
+            $role_name = "";
+        } else {
+            array_push($errors, "Error adding role: ", mysqli_error($conn));
+        }
+    }
+}
+
+// UPDATE
+if (isset($_POST['update_role'])) {
+
+    $id = mysqli_real_escape_string($conn, $_GET['update_id']);
+
+    // receive all input values from the form
+    $role_name = mysqli_real_escape_string($conn, $_POST['role_name']);
+
+    // form validation: ensure that the form is correctly filled ...
+    // by adding (array_push()) corresponding error unto $errors array
+    if (empty($role_name)) {
+        array_push($errors, "Role Name is required");
+    }
+
+    if (count($errors) == 0) {
+        $update = "UPDATE roles set role_name = '$role_name' WHERE role_id ='$id'";
+
+        if (mysqli_query($conn, $update)) {
+            array_push($success, "Update Successful");
+            // clear variables
+            $role_name = "";
+        } else {
+            array_push($errors, "Error updating role: ", mysqli_error($conn));
+        }
+    }
+}
+
+// DELETE
+if (isset($_GET['delete_id'])) {
+    $id = mysqli_real_escape_string($conn, $_GET['delete_id']);
+    $delete = "DELETE FROM roles WHERE role_id='$id'";
+    if (mysqli_query($conn, $delete)) {
+        array_push($success, "Delete successful");
+    } else {
+        array_push($errors, "Delete error: " . mysqli_error($conn));
+    }
+}
+
+?>
+
+<div class="content-body">
+    <?php if (isset($_GET['delete_view'])) {
+        display_success();
+        display_error();
+    }
+
+    $query = "SELECT * FROM roles ORDER BY role_id ASC";
+    $results = mysqli_query($conn, $query);
+
+    ?>
+    <p><b>Roles</b></p>
+    <table>
+        <thead>
+            <tr>
+                <?php isAdmin() ? print '<th>Role ID</th>' : ''; ?>
+                <th>Role Name</th>
+                <?php isAdmin() ? print '<th colspan="2">Action</th>' : ''; ?>
+
+            </tr>
+        </thead>
+        <tbody>
+            <?php
+            while ($row = mysqli_fetch_assoc($results)) {
+                $id = $row['role_id'];
+                $role_name = $row['role_name'];
+            ?>
```

Comment:
As per our DB, role table has 'role description' - so we need to be able to view that